src/public/disease/sections/KnownDrugs/Section.js

```diff
@@ -113,84 +115,119 @@ const headerGroups = [
   })),
 ];
 
-const Section = ({ data, fetchMore, efoId }) => {
-  const pageSize = 10;
-  const [cursor, setCursor] = useState(data.cursor);
+const fetchDrugs = (efoId, cursor, size, freeTextQuery) => {
+  return client.query({
+    query: KNOWN_DRUGS_QUERY,
+    variables: {
+      efoId,
+      cursor,
+      size,
+      freeTextQuery,
+    },
+  });
+};
+
+const Section = ({ efoId }) => {
+  const [loading, setLoading] = useState(true);
+  const [count, setCount] = useState(0);
+  const [cursor, setCursor] = useState(null);
+  const [rows, setRows] = useState([]);
+  const [page, setPage] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
   const [globalFilter, setGlobalFilter] = useState('');
-  const [loading, setLoading] = useState(false);
-  const [rows, setRows] = useState(data.rows.slice(0, pageSize));
+
+  useEffect(
+    () => {
+      fetchDrugs(efoId).then(res => {
```

Comment:
`useEffect` does not allow for async functions to be passed to it. The alternative would be to define the async function inside the `useEffect` callback but I would rather just use a promise.

Reply:
> The alternative would be to define the async function inside the useEffect callback

Exactly! Take a look at: https://github.com/facebook/react/issues/14326. Abramov himself recommends to do just that in that thread.

I've already used the style you are mentioning throughout the code, so let's stick to one way of doing things.

In any case, the async/await syntactic sugar is more modern, and it would be beneficial to drop the old promise chaining in our codebase.

Reply:
As we decided in the meeting, using `then` is all right in here. No promise chaining or anything complex being done asynchronously, so the two ways are similar in cognitive load.